### PR TITLE
chore: Use published `router-bridge`

### PR DESCRIPTION
**Pull request by o0Ignition0o**
part of #491

We have published the `router-bridge` to crates.io, which removes the need for router developers to have nodejs installed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -238,9 +166,11 @@ commands:
...
   build_common_permutations:
     steps:
-      - rust/build:
-          with_cache: false
-          crate: --locked -p apollo-router
+      - run:
+          name: cargo check workspace
+          command: |
+            set -e -o pipefail
+            cargo check --locked
```
Comment (SimonSapin) [inline .circleci/config.yml:173]: Isn't this redundant with the build that's necessary for running tests?

File: .circleci/config.yml [hunk 2]
```diff
@@ -77,87 +75,17 @@ commands:
...
-  linux_prepare_node_env:
-    steps:
-      #TODO[igni]: check for node version before we try to install it
-      - run:
-          name: Install nvm
-          command: |
-            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
-            echo '. ~/.nvm/nvm.sh' >> $BASH_ENV
-      - run:
-          name: Install desired Node.js version
```
Comment (SimonSapin) [inline .circleci/config.yml:114]: Very nice to be able to remove all this!

**Comment by garypen** [comment]
> part of #491
> 
> We have published the `router-bridge` to crates.io, which removes the need for router developers to have nodejs installed.

Does this mean we can get rid of the ASDF stuff in our repo as well now?

**Comment by o0Ignition0o** [comment]
> Does this mean we can get rid of the ASDF stuff in our repo as well now?
oh we could! let s try!

**Review by garypen — APPROVED** [review]
Best PR ever! :)

**Review by Geal — APPROVED** [review]

File: .tool-versions [hunk 3]
```diff
@@ -1,5 +0,0 @@
-# If updating this, you *must* also change the channel in rust-toolchain.toml
-# renovate-automation: rustc version
-rust 1.61.0
```
Comment (abernix) [inline .tool-versions:3]: @o0Ignition0o Curious — this PR was intended to remove the dependency on Node.js, which was indeed great, but I'm curious why you chose to remove `rust` from `.tool-versions` as well?  Would we consider adding it back?  Do we think it's unnecessary?

Comment (SimonSapin) [reply .tool-versions:3]: As far as I know `.tool-versions` is only used by ASDF. In my experience folks writing Rust regularly have `$(which cargo)` pointing to rustup, not ASDF. With this PR, ASDF is not needed at all to contribute to the Router.

**Pull request merged**